docs/sources/flow/reference/components/prometheus.integration.redis.md

```diff
@@ -0,0 +1,104 @@
+---
+# NOTE(rfratto): the title below has zero-width spaces injected into it to
+# prevent it from overflowing the sidebar on the rendered site. Be careful when
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.​integration.redis
+---
+
+# prometheus.integration.redis
+The `prometheus.integration.redis` component embeds
+[redis_exporter](github.com/oliver006/redis_exporter) for collecting metrics from a redis database.
+
+## Usage
+
+```river
+prometheus.integration.redis "LABEL"{
+}
+```
+
+## Arguments
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`redis_addr`                  | `string`   | Address (host and port) of the Redis instance we should connect to. This could be a local Redis instance (e.g. `localhost:6379`), or the address of a remote server. | | yes
+`redis_user`                  | `string`   | User name to use for authentication (Redis ACL for Redis 6.0 and newer).  | | no
+`redis_password`              | `string`   | Password of the redis instance. | | no
+`redis_password_file`         | `string`   | Path of a file containing a passord. If this is defined, it takes precedece over redis_password. | | no
+`redis_password_map_file`     | `string`   | Path of a file containing a JSON object which maps Redis URIs to passwords (e.g. `{"redis://localhost:6379": "sample_password"}`). | | no
+`namespace`                   | `string`   | Namespace for the metrics.  | `"redis"` | no
+`config_command`              | `string`   | What to use for the CONFIG command. | `"CONFIG"` | no
+`check_keys`                  | `string`   | Comma separated list of key-patterns to export value and length/size, searched for with SCAN. | | no
+`check_key_groups`            | `string`   | Comma separated list of LUA regex for grouping keys. When unset, no key groups will be made. | | no
```

Comment:
```suggestion
`check_keys`                  | `string`   | Comma-separated list of key-patterns to export value and length/size, searched for with SCAN. | | no
`check_key_groups`            | `string`   | Comma-separated list of Lua regular expressions (regex) for grouping keys. When unset, no key groups are made. | | no
```

Reply:
Hyphenate compound modifiers before a noun. We use Google Developer Style Guide -- here's the section: https://developers.google.com/style/hyphens#compound-modifiers

Reply:
Good to know - thanks!